data-model/src/main/java/io/micronaut/data/annotation/QueryResultTransformer.java

```diff
@@ -43,4 +43,9 @@
      * @return The column containing JSON view result
      */
     String column() default DATA_COLUMN;
+
+    /**
+     * @return The media type result representation. For example "application/json"
+     */
+    String mediaType();
```

Comment:
Please introduce a custom enum `QueryResultType` with values: `TABULAR` - previous SQL row mapper, `JSON`

Reply:
Not sure what you have in mind, but mapper we use now and JSON will be different as JSON reads from the column and needs column info while for tabular column is obsolete info and we wouldn't add annotation in these cases?

Reply:
The column would be used only for the `JSON` query result

Reply:
Ok, done.